### ISSUE TITLE
Setting `requiresReload` doesn't trigger a reload when the setting is changed programmatically

### ISSUE THREAD

**Issue by arcanistzed** [foundryvtt/foundryvtt#7740]
### User Experience

`requiresReload` implies that a reload (or at least a prompt for a reload) will be triggered when the setting is changed. Not triggering a reload when a setting such as `core.moduleConfiguration` is changed via the API also breaks modules like Find the Culprit which rely on that.

**Comment by Fyorl** [comment]
The problem is that it's not possible to tell which invocation of `ClientSettings#set` is an 'API-only' call, and which is being performed as part of a larger process. For example, when changing settings via the UI, it performs a `ClientSettings#set` to persist the changes. If every setting with `requiresReload` triggered a reload on that call, the UI would not be able to prompt the user, since the page would have already reloaded.

It's still possible for modules that modify these settings to trigger a reload themselves, and we added a helper `debouncedReload` to do just that. It's slightly more inconvenient than before, sure, but it seems necessary to allow various workflows to take some action before reload.

**Comment by arcanistzed** [comment]
This might be how I would approach that: calling `ClientSettings#set` would prompt the user with the Dialog (unless there is already a Dialog for this open) and then only reload after they confirm. This should mean that if multiple `ClientSettings#set` calls happen at once, there would only be one dialog and the user would be able to reload a single time after all of those calls have been processed.

**Comment by Fyorl** [comment]
I don't think it's a good idea for API calls like that to spawn UI prompts as a side effect. It would be weird if `Document#delete` spawned a prompt to the user asking if they were sure they wanted to delete the document, for example. They are separate concerns.

**Comment by arcanistzed** [comment]
Good point. What about an option for `ClientSettings#set` to `promptReload`? When true, it would show a prompt before reloading and otherwise it would just reload without a prompt. The settings config would call `ClientSettings#set` with `promptReload: true`.

**Comment by Fyorl** [comment]
Yeah, inverting the responsibility like that is a potential option.

**Comment by cswendrowski** [comment]
Bumping out of V10
